src/output/workspace-impl.cpp

```diff
@@ -398,6 +398,50 @@ class output_viewport_manager_t
         current_vx = 0;
         current_vy = 0;
     }
+    /**
+     * @param threshold Threshold of the view to be counted 
+     *        on that workspace. 1.0 for 100% visible, 0.1 for 10%
+     *
+     * @return a vector of all the workspaces
+     */
+    std::vector<wf::point_t> get_view_workspaces(wayfire_view view, double threshold)
+    {
+        wf::dimensions_t output_workspaces;
+        std::vector<wf::point_t> view_workspaces;
+        wf::output_t *output = view->get_output();
+        output_workspaces = output->workspace->get_workspace_grid_size();
+
+        wlr_box bounding_box;
+        wf::geometry_t output_relative_geometry;
+        wf::geometry_t intersection;
+        double area;
+        if (threshold != 1.0)
+         {
+             bounding_box = view->get_bounding_box();
+             output_relative_geometry = view->get_output()->get_relative_geometry();
+             intersection = wf::geometry_intersection(bounding_box, output_relative_geometry);
+             area = 1.0 * intersection.width * intersection.height;
+             area /= 1.0 * bounding_box.width * bounding_box.height;
+         }
+
+        for (int row = output_workspaces.width; row-- > 0;)
```

Comment:
1. Please don't use `--` and `>` in the same expression, I can understand what's going on but I don't think it adds anything but makes the code a little more complex.
2. Workspaces are generally numbered from 0 to vwidth-1, it is strange that this function returns them in {1 .. vwidth}